Assets/js/customer.js

```diff
@@ -249,4 +252,73 @@ document.getElementsByClassName("filter--div")[0].addEventListener('click', func
         initialProducts.sort(compare);
         showCarts();
     }
-}); 
\ No newline at end of file
+}); 
+
+
+// Search bar
+// Name Array
+let nameArray = [];
+initialProducts.map(function(cartItem,i){
+    if(!(nameArray.includes(cartItem.productName))){
+        nameArray.push(cartItem.productName);
+    }
+});
+
+// create li and a elements
+for(let i=0; i<nameArray.length; i++){
+    let li = document.createElement('li');
+    li.className = "li-search";
+    document.getElementById("myUL").appendChild(li);
+    let a = document.createElement('a');
+    a.className = "a-search";
+    a.innerText = nameArray[i];
+    li.appendChild(a);
+}
+// Search Suggestions
+document.getElementById("search-input").addEventListener("keyup", function(){
+    // Show Options
+    document.getElementById("myUL").style.display = "block";
+    // If Input Empty hide options
+    if(document.getElementById("search-input").value == ""){
+        document.getElementById("myUL").style.display = "none";
+    }
+
+    let input, filter, ul, li, a, i, txtValue;
+    input = document.getElementById("search-input");
+    filter = input.value.toUpperCase();
+    ul = document.getElementById("myUL");
+    li = ul.getElementsByTagName("li");
+    for (let i = 0; i < li.length; i++) {
+        a = li[i].getElementsByTagName("a")[0];
+        txtValue = a.textContent || a.innerText;
+        if (txtValue.toUpperCase().indexOf(filter) > -1) {
+            li[i].style.display = "";
+        } else {
+            li[i].style.display = "none";
+        }
+    }
+
+    // Select Specific Name from search
+    for(let k=0; k<nameArray.length; k++){
+        document.getElementsByClassName("a-search")[k].addEventListener('click', function(){
```

Comment:
It might be simpler if you add this event listener up on line 273, when you are creating the `a` element.  Then you don't have to loop over nameArray down here and find the `a` again, and so on.

Reply:
done